api/agent/drivers/stats/stats.go

```diff
@@ -130,3 +132,38 @@ func Decimate(maxSamples int, stats []Stat) []Stat {
 	}
 	return stats[:nextEntry] // Return slice of []Stats that was modified with averages
 }
+
+// GenerateLogScaleHistogramBucketsWithRange generates histogram buckets on the log scale between the specified min and max range,
+// such that the min value is in the first bucket and the max in the last.
+func GenerateLogScaleHistogramBucketsWithRange(min, max float64) []float64 {
+	if min <= 0 {
+		logrus.Fatal("cannot generate log scale with non positive domain values")
+	}
+	count := int(math.Ceil(math.Log2(max / min)))
+	var res []float64
+	for i := 0; i <= count; i++ {
```

Comment:
last bucket is bigger than max (eg. min:2 max:10, gets 2,4,8,16) , off by one error.

Reply:
true! I'll change this up to divide from max instead of multiply from min.